.travis.yml

```diff
@@ -19,6 +19,9 @@ script:
 - if [ "$TRAVIS_RUST_VERSION" == "nightly" ]; then
     cargo bench --verbose --no-run;
   fi
+- if [ "$TRAVIS_RUST_VERSION" == "nightly" ]; then
```

Comment:
`conservative_impl_trait` seems to be used only in commented out iterator methods.
`associated_consts` seems to be used for `Key` trait.
`slice_patterns` seems to be used in `Leaf2::shrink_remove` (I actually added this code... Didn't even realise it was only allowed because of feature).
These all can be worked around, so it could be made to work in stable.